### DISCUSSION TITLE
Question regarding KafkaUsers, the user operator, and reconciliation

### DISCUSSION THREAD

**Discussion by mjrepo2** [strimzi/strimzi-kafka-operator#5503] [Q&A]
Hello I am using strzimi 0.20.1, and I am trying to enable access to the kafka cluster with a kafkauser and secret. I have created the kafkauser resource and a secret is provided for me. I need a "username" field for my application in this kafka secret, so I did an edit on the existing secret and wrote a username field. The reconciliation however reverts the secret to its orignal state (what I know is 100% correct and working as intended) but this makes it impossible for me to use this secret.
Basically - is there any way to turn reconciliation off for the user operator? (when I remove it from the kafka.yaml, it just defaults to 120)
if not - Is there any way to get the user operator to include a user name data field?
Or am I missing something?

As usual thanks for any help

**Comment by scholzj** [comment]
Why do you need the username in the secret? The name of the secret is the username, so you should already know it to get to the secret content in the first place.

There is no way to turn the reconciliation off. The operator has to manage things to keep everything working -> that is the main purpose. You can disable the operator by removing the `userOperator: {...}` in the Kafka CR if you want to manage the things your self.
